lib/dialects/abstract/connection-manager.js

```diff
@@ -265,31 +265,23 @@ class ConnectionManager {
       promise = Promise.resolve();
     }
 
-    return promise.then(() =>
-      new Promise((resolve, reject) => {
-        const connectionPromise = this.pool.acquire(options.priority, options.type, options.useMaster);
-        const connectionTimer = timers.setInterval(() => {
-          let evictTimer = false;
-
-          if (connectionPromise.isFulfilled()) {
-            resolve(connectionPromise);
-            debug('connection acquire');
-            evictTimer = true;
-          } else if (this.poolError) {
+    return promise.then(() => {
+      return Promise.race([
+        this.pool.acquire(options.priority, options.type, options.useMaster),
+        new Promise((resolve, reject) =>
+        timers.setTimeout(() => {
+          if (this.poolError) {
             reject(this.poolError);
-            this.poolError = null;
-            evictTimer = true;
-          } else if (connectionPromise.isRejected()) {
-            connectionPromise.catch(reject);
-            evictTimer = true;
           }
-
-          if (evictTimer) {
-            timers.clearInterval(connectionTimer);
-          }
-        }, 0);
-      })
-    );
+        }, 0))
```

Comment:
This event is emitted by pooling library, not internal db layer. Also creating an event handler on each acquire call is not good. With current solution we will have only one event handler for each Sequelize constructor

Basically its same just as you suggested, but instead of creating listener every time, we have a listener per constructor which keep populating `this.poolErr`